server/plugin/hooks.go

```diff
@@ -21,19 +21,25 @@ func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
 		return
 	}
 
-	channel, appErr := p.API.GetChannel(post.ChannelId)
-	if appErr != nil {
-		p.API.LogError("Error occurred while fetching channel by ID. ChannelID: %s. Error: %s", post.ChannelId, appErr.Error())
-		return
-	}
+	if presentInCache := p.dmChannelCache.Has(post.ChannelId); !presentInCache {
+		channel, appErr := p.API.GetChannel(post.ChannelId)
+		if appErr != nil {
+			p.API.LogError("Error occurred while fetching channel by ID. ChannelID: %s. Error: %s", post.ChannelId, appErr.Error())
+			return
+		}
 
-	if channel.Type != model.CHANNEL_DIRECT {
-		return
-	}
+		if channel.Type != model.CHANNEL_DIRECT {
+			return
+		}
```

Comment:
Should a channel that is not a DM be added to the cache?

Reply:
Right now, we are adding only those channels which are DM channels with the VA bot. But now that I think more about it, I think we should add other channels to the cache if we want to solve our performance concerns. I'll make that change.